pipelines/build/common/config_regeneration.groovy

```diff
@@ -372,21 +412,20 @@ class Regeneration implements Serializable {
     */
     def queryAPI(String query) {
         try {
-            def parser = new JsonSlurper()
             def get = new URL(query).openConnection()
 
-            String jenkinsAuth = ""
-            if (jenkinsUsername != "") {
-                jenkinsAuth = "Basic " + new String(Base64.getEncoder().encode("$jenkinsUsername:$jenkinsToken".getBytes()))
+            // Set request credentials if they exist
+            if (jenkinsCreds != "") {
+                def jenkinsAuth = "Basic " + new String(Base64.getEncoder().encode(jenkinsCreds.getBytes()))
+                get.setRequestProperty ("Authorization", jenkinsAuth)
             }
-            get.setRequestProperty ("Authorization", jenkinsAuth)
 
-            def response = parser.parseText(get.getInputStream().getText())
+            def response = new JsonSlurper().parseText(get.getInputStream().getText())
             return response
         } catch (Exception e) {
             // Failed to connect to jenkins api or a parsing error occurred
-            context.println "[ERROR] Failure on jenkins api connection or parsing.\nError: ${e}"
-            currentBuild.result = "FAILURE"
+            context.println "[ERROR] Failure on jenkins api connection or parsing.\n${e}"
+            throw new Exception()
```

Comment:
pass e into the exception for chaining messages

Reply:
Not sure what you mean?